src/Framework/Framework/Compilation/Directives/BaseTypeDirectiveCompiler.cs

```diff
@@ -107,19 +108,15 @@ IEnumerable<DothtmlDirectiveNode> propertyDirectives
                 ? new ResolvedTypeDescriptor(createdTypeInfo)
                 : null;
         }
-        
+
         /// <summary>
         /// Gets the default type of the wrapper for the view.
         /// </summary>
         private ITypeDescriptor GetDefaultWrapperType()
-        {
-            if (fileName.EndsWith(".dotcontrol", StringComparison.Ordinal))
-            {
-                return new ResolvedTypeDescriptor(typeof(DotvvmMarkupControl));
-            }
+            => new ResolvedTypeDescriptor(IsMarkupControl() ? typeof(DotvvmMarkupControl) : typeof(DotvvmView));
```

Comment:
Determining the wrapper type based on file extension is fragile. Now everything worked even when the base type was incorrectly inferred as DotvvmView (#1479), so people can use dothtml or html extension for controls. I think it would be better to determine if a page is control based on the registration in dotvvm configuration:

```
configuration.Markup.Controls.Any(c => fileName.Equals(c.Src, StringComparison.OrdinalIgnoreCase))
```

However, that would run into problem every time a new control is added in VS Extension, so I'd use an "extension OR configuration" condition in there (this class is overwritten anyway, so if this method is abstract it should be easy, right?)